conan/tools/sbom/cyclonedx.py

```diff
@@ -198,4 +197,10 @@ def _calculate_licenses(component):
         for l in licenses
     ]
 
+def _calculate_bomref(component):
+    user, channel = component.ref.user, component.ref.channel
+    user_channel = f"?user={user}?channel={channel}" if user and channel else f"?user={user}" if user else ""
+    return f"pkg:conan/{component.name}@{component.ref.version}?rref={component.ref.revision}{user_channel}"
```

Comment:
```suggestion
    user = "&user={component.ref.user}" if component.ref.user else ""
    channel = "&channel={component.ref.channel}" if component.ref.channel else ""
    return f"pkg:conan/{component.name}@{component.ref.version}?rref={component.ref.revision}{user}{channel}"
```

Note the `&` to make it more URL like, and the fact that we don't check user AND channel at the same time. If there's a channel, we have already added a user by construction of the ref object itself

Reply:
Question from my ignorance. Does the extra ``?user=xxx`` fields could break somehow tools that use the ``bom_ref`` for analysis? Or those tools understand they want to check for ``pkg@version`` and probably discard the optional user/channel fields?

Reply:
It shouldn't break anything. The bom-ref is an internal value within the SBOM used to reference other packages. Neither the format nor the content of the bom-ref is defined. Here is the definition from the standard:
1.6:
```
An optional identifier which can be used to reference the component elsewhere in the BOM. 
Every bom-ref must be unique within the BOM.
Value SHOULD not start with the BOM-Link intro 'urn:cdx:' to avoid conflicts with BOM-Links.
```
https://cyclonedx.org/docs/1.6/json/#metadata_component_bom-ref
1.4:
```
An optional identifier which can be used to reference the component elsewhere in the BOM. 
Every bom-ref MUST be unique within the BOM.
```
https://cyclonedx.org/docs/1.4/json/#metadata_component_bom-ref